media_driver/linux/common/vp/ddi/media_libva_vp.c

```diff
@@ -1174,15 +1174,15 @@ DdiVp_SetProcPipelineParams(
 
     // Update fwd and bkward ref frames: Required for Advanced processing - will be supported in the future
 
-    pVpHalSrcSurf->uFwdRefCount  = pPipelineParam->num_forward_references;
+    pVpHalSrcSurf->uFwdRefCount  = pPipelineParam->num_backward_references;
 
-    vaStatus = DdiVp_UpdateProcPipelineForwardReferenceFrames(pVpCtx, pVaDrvCtx, pVpHalSrcSurf, pPipelineParam);
-    DDI_CHK_RET(vaStatus, "Failed to update forward reference frames!");
+    vaStatus = DdiVp_UpdateProcPipelineFutureReferenceFrames(pVpCtx, pVaDrvCtx, pVpHalSrcSurf, pPipelineParam);
+    DDI_CHK_RET(vaStatus, "Failed to update future reference frames!");
 
-    pVpHalSrcSurf->uBwdRefCount  = pPipelineParam->num_backward_references;
+    pVpHalSrcSurf->uBwdRefCount  = pPipelineParam->num_forward_references;
```

Comment:
As far as I know, forward predict usually means that use previous sequences to predict  current value. Maybe we need to change vphal to align with VAAPI.

Reply:
This WA fix may be confusing.

Reply:
Assign forward to backward and backward to forward. Just correct DDI level mapping is confused. We need to change behavior in vphal level for formal fix.